ui/src/services/core/control-service/control-service.ts

```diff
@@ -0,0 +1,171 @@
+import { socket } from '@/api/socket'
+import { TransactionService } from '@/services/core/transaction-service/transaction-service'
+
+import { KEYBOARD_KEYS_MAP } from '@/constants/keyboard-keys-map'
+
+import type { SendTwoWayArgs, TouchDownArgs, TouchMoveArgs, TouchMoveIosArgs } from './types'
+
+class ControlService {
```

Comment:
I think at this point it makes sense for the ControlService to know everything about the device that it's controlling: name, serial and the channel. We should instantiate it after we have a connection and invited the device into a channel and expose it using context or somewhere else in the app state. That way we won't need to pass deviceChannel to every single method and the isDeviceIos flag

And rename it to DeviceControlService